### PR TITLE
Update comments: Source Match feature and poll period selection

### PR DESCRIPTION
**Pull request by abtink**
This commit adds/updates comments. It contains no code/logic change.

- Adds comments to describe expected behavior and use of source match
  feature and the related radio platform functions.

- Updates comments to describe the selection of poll period and the
  `SetAssignPollPeriod()`.

### CODE REVIEW

File: src/core/thread/mesh_forwarder.hpp [hunk 1]
```diff
@@ -131,23 +131,32 @@ class MeshForwarder
...
+     * This method sets a user-specified Data Poll period.
      *
-     * @param[in]  aPeriod  The Data Poll period in milliseconds.
+     * If the value is set to zero, then the poll interval is controlled/chosen by the OpentThread stack.
+     * If the user has provided a non-zero poll period, the user value has higher priority and is used in place of
+     * poll period values chosen internally by OpenThread implementation (except during attachment).
+     *
+     * Initial/Default value for "assign poll period" is zero.
+     *
+     * @param[in]  aPeriod  The Data Poll period in milliseconds. Value zero
```
Comment (LuDuda) [inline src/core/thread/mesh_forwarder.hpp]: Value zero... ?

File: src/core/thread/mesh_forwarder.hpp [hunk 2]
```diff
@@ -131,23 +131,32 @@ class MeshForwarder
...
     uint32_t GetAssignPollPeriod(void);
 
     /**
-     * This method sets the Data Poll period.
+     * This method sets/updates the Data Poll period.
+     *
+     * If the user has provided a non-zero assign poll period (@sa SetAssignPollPeriod), the user value has higher
+     * priority and is used in place of @p aPeriod (except during attachment).
      *
      * @param[in]  aPeriod  The Data Poll period in milliseconds.
```
Comment (LuDuda) [inline src/core/thread/mesh_forwarder.hpp]: Should we specify that value of 0 is incorrect. Or it is correct?

File: src/core/thread/mesh_forwarder.hpp (hunk 1 above)
Comment (abtink) [inline src/core/thread/mesh_forwarder.hpp]: Good catch! Unfinished sentence... will remove it... thanks...

File: src/core/thread/mesh_forwarder.hpp (hunk 2 above)
Comment (abtink) [inline src/core/thread/mesh_forwarder.hpp]: I was also considering this. 
I would suggest keeping value zero as a valid input but treat it to mean to stop periodic polling. Thoughts?

Comment (jwhui) [inline src/core/thread/mesh_forwarder.hpp]: `SetPollPeriod()` is currently internal to OpenThread (maybe something to add in the comments).  I don't think there is any path through the OpenThread API that would result in passing `0` to this.  If we want to completely disable polling while in the rx-off-when-idle mode, we should add that feature in a separate PR.  Is there a case for that?  The user value could always be set to a large value.

**Comment by abtink** [comment]
Thanks @jwhui  I updated the PR and also rebased it on master...

**Pull request merged**